ingestion/feat.ingestion/Handlers/FAA/FaaIngestionHandler.cs

```diff
@@ -89,70 +100,503 @@ public override async Task<bool> IngestAsync(CancellationToken cancellationToken
+                {
+                    options.ColumnPrimaryKeyExpression = a => new
+                    {
+                        a.ApprenticeshipId,
+                        a.Postcode,
+                        a.AddressLine1
+                    };
+                }, cancellationToken);
+            }
+            
+            Console.WriteLine($"Find An Apprenticeship staging ingestion complete.");
             return true;
-
         }
         catch (Exception ex)
         {
             Console.WriteLine($"Failed to ingest Find An Apprenticeship data: {ex.Message}");
             return false;
         }
     }
-}
+    
+    public override async Task<bool> SyncAsync(CancellationToken cancellationToken)
+    {
+        var apprenticeships = dbContext.Set<Database.Apprenticeship>()
+            .Include(apprenticeship => apprenticeship.Addresses)
+            .ToList();
+
+        if (apprenticeships.Count == 0)
+        {
+            Console.WriteLine("No data found in staging tables.");
+            return false;
+        }
+
+        Console.WriteLine($"Processing {apprenticeships.Count} apprenticeships from staging...");
+
+        await using var transaction = await dbContext.Database.BeginTransactionAsync(cancellationToken);
+        
+        // PROVIDER
+        
+        var providers = apprenticeships
+            .GroupBy(a => a.Ukprn)
+            .Select(a => new Provider
+        {
+            Created = DateTime.UtcNow,
+            Name = a.First().ProviderName ?? "Unknown provider",
+            Ukprn = a.Key.ToString()
+        }).ToList();
+        
+        await dbContext.BulkSynchronizeAsync(providers, options =>
+        {
+            options.IgnoreOnSynchronizeUpdateExpression = p => new
+            {
+                p.Id,
+                p.Created
+            };
+            options.ColumnPrimaryKeyExpression = p => p.Ukprn;
+        }, cancellationToken);
+
+        var providerLookup = dbContext.Set<Provider>()
+            .ToDictionary(p => p.Ukprn!, p => p.Id);
+
+        Console.WriteLine($"Providers synchronized: {providers.Count}");
+        
+        // ENTRY
+        
+        var entries = apprenticeships.Select(a =>
+        {
+            var providerId = providerLookup[a.Ukprn.ToString()];
+            return new Entry
+            {
+                Created = DateTime.UtcNow,
+                ProviderId = providerId,
+                Reference = a.VacancyReference!,
+                Title = a.Title!,
+                Description = a.Description,
+                FlexibleStart = a.StartDate == null,
+                AttendancePattern = MapCourseHours(a.HoursPerWeek),
+                Url = a.ApplicationUrl ?? string.Empty,
+                SourceSystem = SourceSystem.FAA,
+                Type = EntryType.Apprenticeship,
+                Level = StudyTime.Daytime // TODO: Check
+            };
+        }).ToList();
+        
+        await dbContext.BulkSynchronizeAsync(entries, options =>
```

Comment:
This probably should be a BulkUpsert, rather than a Synchronize, as this would delete entries created outside of FAA unless you can filter it to only FAA created entries